### ISSUE TITLE
XPathNavigator.InnerXml throws when replacement should be a text node

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/corefx#20000]
When the `XPathNavigator.InnerXml` is called with a string that consists only of text (not surrounded by XML tags), it throws an `XmlException`: `Data at the root level is invalid. Line 1, position 1.`. This does not happen on .NET Framework 4.7 (I'm on Creators Update, 15063.296).

This is a fairly straightforward repro:

```C#
XmlDocument xmlDoc = new XmlDocument();

xmlDoc.LoadXml(@"<Product>
  <Name>Book</Name>
</Product>");

XPathNavigator nav = xmlDoc.CreateNavigator();
XPathExpression expr = nav.Compile("/Product/Name");

XPathNodeIterator iter = nav.Select(expr);
iter.MoveNext();

Console.WriteLine(iter.Current.InnerXml);

iter.Current.InnerXml = "Car";
```

Changing the assignment to
```C#
iter.Current.InnerXml = "<Car />";
```
appears to work fine.

[ConsoleApp1.zip](https://github.com/dotnet/corefx/files/1014643/ConsoleApp1.zip) is a standalone repro project. Repro steps:

```
S:\work\innerxml_assignment\ConsoleApp1\ConsoleApp1>dotnet restore
  Restoring packages for S:\work\innerxml_assignment\ConsoleApp1\ConsoleApp1\ConsoleApp1.csproj...
...

S:\work\innerxml_assignment\ConsoleApp1\ConsoleApp1>dotnet run -f net46
Book
Car

S:\work\innerxml_assignment\ConsoleApp1\ConsoleApp1>dotnet run -f netcoreapp1.0
Book

Unhandled Exception: System.Xml.XmlException: Data at the root level is invalid. Line 1, position 1.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ParseRootLevelWhitespace()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XPath.XPathNavigator.BuildSubtree(XmlReader reader, XmlWriter writer)
   at System.Xml.XPath.XPathNavigator.AppendChild(XmlReader newChild)
   at ConsoleApp1.Program.Main(String[] args)
```

I see the same issue when targeting `netcoreapp1.0` and `netcoreapp1.1`, but, confusingly, not `netcoreapp2.0`. Versions of the `System.Xml.*` package references are 4.3.0 across all TFMs (and I see the same behavior with 4.0.1).

<details>
<summary>dotnet --info</summary>

```
S:\work\innerxml_assignment\ConsoleApp1\ConsoleApp1>dotnet --info
.NET Command Line Tools (2.0.0-preview1-005977)

Product Information:
 Version:            2.0.0-preview1-005977
 Commit SHA-1 hash:  414cab8a0b

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0-preview1-005977\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview1-002111-00
  Build    : 1ff021936263d492539399688f46fd3827169983
```
</details>

I ran into this while investigating https://github.com/Microsoft/msbuild/issues/2105.

**Comment by danmosemsft** [comment]
@sepidehMS can you evaluate

**Comment by danmosemsft** [comment]
>I see the same issue when targeting netcoreapp1.0 and netcoreapp1.1, but, confusingly, not netcoreapp2.0. Versions of the System.Xml.* package references are 4.3.0 across all TFMs

@rainersigwald I'm confused by how the same System.Xml dll can get loaded in those 3 cases. Are you sure (eg does procmon show it). We do not have a concept of targeting a different TFM using the same binaries, as far as I know. I would not expect that to work. I suspect this is just a bug that was fixed in 2.0.

**Comment by rainersigwald** [comment]
What I meant by that is that I'm not changing the NuGet package reference. I do not understand assembly loading with the shared framework. Taking wild guesses, maybe something moved into the shared framework (at a higher version with a change in behavior here) and the package references are now unnecessary in netcoreapp2.0?

Here are procmon traces filtered to `system.xml*` of the three situations: [SystemXmlTraces.zip](https://github.com/dotnet/corefx/files/1019527/SystemXmlTraces.zip).

This might also explain why running MSBuild when packaged in the 2.0-preview SDK still shows problems: there, explicit System.Xml.* packages are mentioned in deps.json.

**Comment by karelz** [comment]
Marking 2.0, until proven otherwise. Looks like compat issue against Desktop.

**Comment by sepidehMS** [comment]
@rainersigwald Like you mentioned, what is happening is that on netcoreapp2.0, all the packages you are loading are being overriden by the newer packages brought in by Microsoft.NET.Sdk, since all those packages are now part of that, and the package references are no longer necessary. This seems like a bug in netcoreapp1.0 and 1.1 that got fixed in 2.0. So, I'm closing this.
